code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -685,27 +705,32 @@ Pass a positive integer as an argument to override a bot's default speed.
 
 /mob/living/simple_animal/bot/proc/get_next_patrol_target()
 	// search the beacon list for the next target in the list.
-	for(var/obj/machinery/navbeacon/NB in GLOB.navbeacons["[z]"])
-		if(NB.location == next_destination) //Does the Beacon location text match the destination?
-			destination = new_destination //We now know the name of where we want to go.
-			patrol_target = NB.loc //Get its location and set it as the target.
-			next_destination = NB.codes["next_patrol"] //Also get the name of the next beacon in line.
-			return TRUE
+	var/temporary_z = SSmapping.levels_by_trait(ZTRAIT_STATION)
+	for(var/ZYes in temporary_z)
```

Comment:
what is this awful var name, also if temporary_z is unused don't define the list var